File: fastapidemo/points.py
from fastapi import FastAPI
from fastapi.responses import JSONResponse
from fastapi.middleware.cors import CORSMiddleware

app = FastAPI()

# Configure CORS
app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],  # Allows all origins
    allow_credentials=True,
    allow_methods=["*"],  # Allows all methods
    allow_headers=["*"],  # Allows all headers
)

@app.get("/getPoints")
async def get_points():
    data = {
        "sports": [
            {
                "name": "Riverwalk Stadium",
                "lon": -86.3078,
                "lat": 32.3802
            },
            {
                "name": "Cramton Bowl",
                "lon": -86.2988,
                "lat": 32.3835
            }
        ],
        "music": [
            {
                "name": "Capitol Sounds",
                "lon": -86.3021,
                "lat": 32.3773
            },
            {
                "name": "Montgomery Performing Arts Centre",
                "lon": -86.3080,
                "lat": 32.3748
            }
        ]
    }
    return JSONResponse(content=data, media_type="application/json")

File: mapservices/points/points.go
// Service url takes URLs, generates random short IDs, and stores the URLs in a database.
package points

import (
	"context"
)

type Location struct {
	Name string  `json:"name"`
	Lon  float64 `json:"lon"`
	Lat  float64 `json:"lat"`
}

type Locations struct {
	Sports []Location `json:"sports"`
	Music  []Location `json:"music"`
}

// Shorten shortens a URL.
//encore:api public method=GET path=/getPoints
func getPoints(ctx context.Context) (*Locations, error) {

	data := Locations{
        Sports: []Location{
            {
                Name: "Riverwalk Stadium",
                Lon:  -86.3078,
                Lat:  32.3802,
            },
            {
                Name: "Cramton Bowl",
                Lon:  -86.2988,
                Lat:  32.3835,
            },
        },
        Music: []Location{
            {
                Name: "Capitol Sounds",
                Lon:  -86.3021,
                Lat:  32.3773,
            },
            {
                Name: "Montgomery Performing Arts Centre",
                Lon:  -86.3080,
                Lat:  32.3748,
            },
        },

	}	
	return &data, nil
}
